src/navigationStore.js

```diff
@@ -450,25 +463,24 @@ class NavigationStore {
   nextState = (state, cmd) => (this.reducer ? this.reducer(state, cmd) : reducer(state, cmd));
 
   dispatch = (cmd) => {
-    this.setState(this.nextState(this.state, cmd));
+    this.engineDispatch(cmd);
   };
 
-  @action setState = async (newState) => {
+  setState = async (newState) => {
     // don't allow null state
     if (!newState) {
       return;
     }
     const state = getActiveState(newState);
     // avoid double actions
-    if (isEqual(state.params, this._currentParams) && state.routeName === this.currentScene) {
+    if (isEqual(state.params, this.currentParams) && state.routeName === this.currentScene) {
       return;
     }
     const currentScene = this.currentScene;
     const prevScene = this.prevScene;
     this._state = newState;
```

Comment:
I still have this maintaining its own state in here just because I haven't taken the time to see what else we can rip out of here with moving the engines out to their own files.